app/services/importer/get_api_fields.rb

```diff
@@ -7,50 +9,107 @@ module Importer
   class GetApiFields < ApplicationTask
     include FaradayHelpers
 
-    def initialize
-      # @job = job
-      # @url = @job.api_url
-      @url = "https://boards-api.greenhouse.io/v1/boards/cleoai/jobs/4628944002"
-      # @url = "https://boards-api.greenhouse.io/v1/boards/cleoai/jobs/7301308002"
-      # @url = "https://boards-api.greenhouse.io/v1/boards/monzo/jobs/6076740"
-      # @url = "https://boards-api.greenhouse.io/v1/boards/axios/jobs/6009256"
-      # @url = "https://boards-api.greenhouse.io/v1/boards/11fs/jobs/4060453101"
-      # @url = "https://boards-api.greenhouse.io/v1/boards/forter/jobs/7259821002"
-      # @url = "https://boards-api.greenhouse.io/v1/boards/cleoai/jobs/4628944002"
-      @url += "?questions=true&location_questions=true&demographic_questions=true&&compliance=true&pay_transparency=true"
-      # @ats_sections = %w[main_fields custom_fields demographic_questions eeoc_fields data_compliance security_code_fields]
-      @fields = {}
+    def initialize(data, options = {})
+      @data = data
+      @data_source = options[:data_source]
+      @sections = options[:sections] || [:core]
+      @standard_fields = options[:standard_fields] || {}
+      @types = options[:types] || {}
+      @fields = []
       @errors = false
     end
 
     def call
       return unless processable
 
       process
-    rescue StandardError => e
-      Rails.logger.error "Error running GetFormFields: #{e.message}"
-      nil
     end
 
     private
 
     def processable
-      @url # && @job
+      @data
     end
 
     def process
-      p "Hello from GetApiFields!"
+      build_fields
+      log_and_return_fields
+    end
+
+    def add_section_to_fields
+      @fields << @section_fields
+    end
+
+    def build_fields
+      @sections.each do |section|
+        generate_section(section)
+        build_questions(section)
+        add_section_to_fields
+      end
+    end
+
+    def build_questions(section)
+      fetch_questions(section)&.each do |question|
+        @question = question
+        @section_fields[:questions] << create_question
+      end
+    end
+
+    def core_details = @standard_fields[question_id] || {}
 
-      json = fetch_json(@url)
-      return unless json
+    def create_question
+      {
+        attribute: generate_attribute_from_label(question_label),
+        required: question_required?,
+        label: question_label,
+        description: question_description,
+        fields: fetch_question_fields
+      }.merge(core_details)
+    end
+
+    def fetch_question_fields
+      question_fields&.map do |field|
+        {
+          id: field_id(field),
+          type: standardize_type(field_type(field)),
+          max_length: field_max_length(field),
+          options: field_options(field).map do |option|
```

Comment:
Could you extract the map logic outside in a method options(field) or a method options_builder(field) ?